src/Illuminate/Support/ServiceProvider.php

```diff
@@ -77,12 +89,23 @@ protected function loadRoutesFrom($path)
      *
      * @param  string  $path
      * @param  string  $namespace
+     * @param  bool  $publish
      * @return void
      */
-    protected function loadViewsFrom($path, $namespace)
+    protected function loadViewsFrom($path, $namespace, $publish = false)
     {
-        if (is_dir($appPath = $this->app->resourcePath().'/views/vendor/'.$namespace)) {
-            $this->app['view']->addNamespace($namespace, $appPath);
+        if (! $this->viewNamespacesRegistered) {
```

Comment:
(Although now that I look at it, this should probably be pulled into a `registerDefaultViewNamespaces` method.